allennlp/data/image_loader.py

```diff
@@ -1,69 +1,195 @@
-        config: Optional[DetectronInput] = None,
-    ):
-        if config is None:
-            pipeline = detectron.get_pipeline_from_flat_parameters(
-                make_copy=False, fp=DetectronFlatParameters()
+        image_backend: str = None,
+        resize: bool = True,
+        normalize: bool = True,
+        min_size: int = 800,
+        max_size: int = 1333,
+        pixel_mean: Tuple[float, float, float] = (0.485, 0.456, 0.406),
+        pixel_std: Tuple[float, float, float] = (0.229, 0.224, 0.225),
+        size_divisibility: int = 32,
+        **kwargs
+    ) -> None:
+        super().__init__(size_divisibility=size_divisibility, **kwargs)
+        if image_backend is not None:
+            torchvision.set_image_backend(image_backend)
+        self.resize = resize
+        self.normalize = normalize
+        self.min_size = min_size
+        self.max_size = max_size
+        self.pixel_mean = pixel_mean
+        self.pixel_std = pixel_std
+
+    @overrides
+    def load(self, filename: OnePath) -> FloatTensor:
+        image = torchvision.io.read_image(filename).float().to(self.device) / 256
+        if self.normalize:
+            mean = torch.as_tensor(self.pixel_mean, dtype=image.dtype, device=self.device).view(
+                -1, 1, 1
             )
-        elif isinstance(config, DetectronConfig):
-            pipeline = detectron.get_pipeline(make_copy=False, **config._asdict())
-        elif isinstance(config, DetectronFlatParameters):
-            pipeline = detectron.get_pipeline_from_flat_parameters(
-                make_copy=False, **config._asdict()
+            std = torch.as_tensor(self.pixel_std, dtype=image.dtype, device=self.device).view(
+                -1, 1, 1
             )
-        else:
-            raise ValueError("Unknown type of `config`")
-
-        self.mapper = pipeline.mapper
-
-    def load(self, filenames: ManyPaths) -> ImagesWithSize:
-        images = [{"file_name": str(f)} for f in filenames]
-        images = [self.mapper(i) for i in images]
-        images = ImageList.from_tensors([image["image"] for image in images])
-        return (images.tensor.float() / 256, torch.tensor(images.image_sizes, dtype=torch.int32))  # type: ignore
+            image = (image - mean) / std
+        if self.resize:
+            # Adapted from https://github.com/pytorch/vision/blob/
+            # 4521f6d152875974e317fa247a633e9ad1ea05c8/torchvision/models/detection/transform.py#L36.
+            min_size = min(image.shape[-2:])
+            max_size = max(image.shape[-2:])
+            scale_factor = self.min_size / min_size
+            if max_size * scale_factor > self.max_size:
+                scale_factor = self.max_size / max_size
+            image = torch.nn.functional.interpolate(
+                image[None],
+                scale_factor=scale_factor,
+                mode="bilinear",
+                recompute_scale_factor=True,
+                align_corners=False,
+            )[0]
```

Comment:
In order to use these particular transforms directly from torchvision we will need to load a batch of images into an `ImageList`  type of object, instead of one big tensor.